File: leetcode/lc-us/cpp/2244.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int minimumRounds(vector<int> &tasks) {
        unordered_map<int, int> M;
        for (int x: tasks) M[x]++;
        int res = 0;
        for (auto[k, v]: M) {
            if (v == 1) return -1;
            res += (v + 2) / 3;
        }
        return res;
    }
};

File: leetcode/lc-us/py/2244.py
from collections import Counter

from typing import List


class Solution:
    def minimumRounds(self, tasks: List[int]) -> int:
        cnt = Counter(tasks)
        res = 0
        for _, freq in cnt.items():
            if freq == 1:
                return -1
            res += (freq + 2) // 3
        return res
